### PR TITLE
Migrate to eprint/eprintln macros where appropriate.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

File: src/librustdoc/markdown.rs [hunk 1]
```diff
@@ -75,20 +73,15 @@ pub fn render(input: &str, mut output: PathBuf, matches: &getopts::Matches,
...
         Ok(f) => f
     };
 
     let (metadata, text) = extract_leading_metadata(&input_str);
     if metadata.is_empty() {
-        let _ = writeln!(
-            &mut io::stderr(),
-            "rustdoc: invalid markdown file: no initial lines starting with `# ` or `%`"
-        );
+        eprintln("rustdoc: invalid markdown file: no initial lines starting with `# ` or `%`");
```
Comment (kennytm) [inline src/librustdoc/markdown.rs]: Missing `!`.

```
[00:34:43] error[E0423]: expected function, found macro `eprintln`
[00:34:43]   --> /checkout/src/librustdoc/markdown.rs:84:9
[00:34:43]    |
[00:34:43] 84 |         eprintln("rustdoc: invalid markdown file: no initial lines starting with `# ` or `%`");
[00:34:43]    |         ^^^^^^^^ did you mean `eprintln!(...)`?
```

File: src/librustdoc/externalfiles.rs [hunk 2]
```diff
@@ -70,17 +70,14 @@ pub fn load_string<P: AsRef<Path>>(file_path: P) -> Result<String, LoadStringErr
...
         return Err(LoadStringError::ReadFail);
     }
     match str::from_utf8(&contents) {
         Ok(s) => Ok(s.to_string()),
         Err(_) => {
-            let _ = writeln!(&mut io::stderr(),
-                             "error reading `{}`: not UTF-8",
-                             file_path.display());
+            let _ = eprintln!("error reading `{}`: not UTF-8",
+                              file_path.display());
```
Comment (kennytm) [inline src/librustdoc/externalfiles.rs]: Remove the `let _ = `. (Same for a few lines above)

Also, the `use std::io;` is no longer needed.

```
[00:34:44] error: unused import: `std::io`
[00:34:44]   --> /checkout/src/librustdoc/externalfiles.rs:13:5
[00:34:44]    |
[00:34:44] 13 | use std::io;
[00:34:44]    |     ^^^^^^^
[00:34:44]    |
```

File: src/librustdoc/markdown.rs [hunk 3]
```diff
@@ -75,20 +73,15 @@ pub fn render(input: &str, mut output: PathBuf, matches: &getopts::Matches,
 
     let mut out = match File::create(&output) {
         Err(e) => {
-            let _ = writeln!(&mut io::stderr(),
-                             "rustdoc: {}: {}",
-                             output.display(), e);
+            let _ = eprintln!("rustdoc: {}: {}", output.display(), e);
```
Comment (kennytm) [inline src/librustdoc/markdown.rs]: Remove `let _ = `.

File: src/librustdoc/markdown.rs [hunk 4]
```diff
@@ -138,9 +131,8 @@ pub fn render(input: &str, mut output: PathBuf, matches: &getopts::Matches,
 
     match err {
         Err(e) => {
-            let _ = writeln!(&mut io::stderr(),
-                             "rustdoc: cannot write to `{}`: {}",
-                             output.display(), e);
+            let _ = eprintln!("rustdoc: cannot write to `{}`: {}",
+                              output.display(), e);
```
Comment (kennytm) [inline src/librustdoc/markdown.rs]: Remove `let _ = `

File: src/librustdoc/test.rs [hunk 5]
```diff
@@ -479,11 +479,10 @@ impl Collector {
                 found = entry.remove_item(&test).is_some();
             }
             if !found {
-                let _ = writeln!(&mut io::stderr(),
-                                 "WARNING: {} Code block is not currently run as a test, but will \
-                                  in future versions of rustdoc. Please ensure this code block is \
-                                  a runnable test, or use the `ignore` directive.",
-                                 name);
+                let _ = eprintln!("WARNING: {} Code block is not currently run as a test, but will \
```
Comment (kennytm) [inline src/librustdoc/test.rs]: Remove `let _ =`

File: src/librustdoc/markdown.rs [hunk 6]
```diff
@@ -10,8 +10,6 @@
 
 use std::default::Default;
 use std::fs::File;
-use std::io::prelude::*;
-use std::io;
```
Comment (kennytm) [inline src/librustdoc/markdown.rs]: The `std::io::Write` trait is still needed somewhere.

```
[00:34:47] error[E0599]: no method named `write_fmt` found for type `&mut std::fs::File` in the current scope
[00:34:47]    --> /checkout/src/librustdoc/markdown.rs:97:15
[00:34:47]     |
[00:34:47] 97  |       let err = write!(
[00:34:47]     |  _______________^
[00:34:47] 98  | |         &mut out,
[00:34:47] 99  | |         r#"<!DOCTYPE html>
[00:34:47] 100 | | <html lang="en">
[00:34:47] ...   |
[00:34:47] 129 | |         after_content = external_html.after_content,
[00:34:47] 130 | |         );
[00:34:47]     | |_________^
[00:34:47]     |
[00:34:47]     = help: items from traits can only be used if the trait is in scope
[00:34:47]     = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
[00:34:47]             candidate #1: `use std::io::Write;`
[00:34:47]     = note: this error originates in a macro outside of the current crate
```

File: src/librustdoc/externalfiles.rs [hunk 7]
```diff
@@ -9,8 +9,6 @@
 // except according to those terms.
 
 use std::fs::File;
-use std::io::prelude::*;
-use std::io;
```
Comment (kennytm) [inline src/librustdoc/externalfiles.rs]: Looks like this needs `std::io::Read` as well...

```
[00:33:33] error[E0599]: no method named `read_to_end` found for type `std::fs::File` in the current scope
[00:33:33]   --> /checkout/src/librustdoc/externalfiles.rs:69:43
[00:33:33]    |
[00:33:33] 69 |                       .and_then(|mut f| f.read_to_end(&mut contents));
[00:33:33]    |                                           ^^^^^^^^^^^
[00:33:33]    |
[00:33:33]    = help: items from traits can only be used if the trait is in scope
[00:33:33]    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
[00:33:33]            candidate #1: `use std::io::Read;`
```

File: src/test/run-pass/backtrace-debuginfo.rs [hunk 8]
```diff
@@ -163,7 +161,7 @@ fn main() {
     let args: Vec<String> = env::args().collect();
     if args.len() >= 2 {
         let case = args[1].parse().unwrap();
-        writeln!(&mut io::stderr(), "test case {}", case).unwrap();
+        eprintln!("test case {}", case).unwrap();
```
Comment (kennytm) [inline src/test/run-pass/backtrace-debuginfo.rs]: Unnecessary `unwrap()`.

```
[00:47:00] ---- [run-pass] run-pass/backtrace-debuginfo.rs stdout ----
[00:47:00] 	
[00:47:00] error: compilation failed!
[00:47:00] status: exit code: 101
[00:47:00] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/run-pass/backtrace-debuginfo.rs" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/run-pass" "--target=x86_64-unknown-linux-gnu" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/run-pass/backtrace-debuginfo.stage2-x86_64-unknown-linux-gnu" "-Crpath" "-O" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-g" "-Cllvm-args=-enable-tail-merge=0" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/run-pass/backtrace-debuginfo.stage2-x86_64-unknown-linux-gnu.run-pass.libaux"
[00:47:00] stdout:
[00:47:00] ------------------------------------------
[00:47:00] 
[00:47:00] ------------------------------------------
[00:47:00] stderr:
[00:47:00] ------------------------------------------
[00:47:00] error[E0599]: no method named `unwrap` found for type `()` in the current scope
[00:47:00]    --> /checkout/src/test/run-pass/backtrace-debuginfo.rs:164:41
[00:47:00]     |
[00:47:00] 164 |         eprintln!("test case {}", case).unwrap();
[00:47:00]     |                                         ^^^^^^
[00:47:00] 
[00:47:00] error: aborting due to previous error
[00:47:00] 
[00:47:00] 
[00:47:00] ------------------------------------------
[00:47:00] 
[00:47:00] thread '[run-pass] run-pass/backtrace-debuginfo.rs' panicked at 'explicit panic', /checkout/src/tools/compiletest/src/runtest.rs:2433:8
[00:47:00] note: Run with `RUST_BACKTRACE=1` for a backtrace.
[00:47:00] 
[00:47:00] 
[00:47:00] failures:
[00:47:00]     [run-pass] run-pass/backtrace-debuginfo.rs
[00:47:00] 
[00:47:00] test result: FAILED. 2761 passed; 1 failed; 8 ignored; 0 measured; 0 filtered out
```

**Comment by frewsxcv** [comment]
finally got a green checkmark here

File: src/tools/tidy/src/lib.rs [hunk 9]
```diff
@@ -33,10 +33,9 @@ macro_rules! t {
 
 macro_rules! tidy_error {
     ($bad:expr, $fmt:expr, $($arg:tt)*) => ({
-        use std::io::Write;
         *$bad = true;
-        write!(::std::io::stderr(), "tidy error: ").expect("could not write to stderr");
-        writeln!(::std::io::stderr(), $fmt, $($arg)*).expect("could not write to stderr");
+        eprint!("some tidy checks failed");
```
Comment (ollie27) [inline src/tools/tidy/src/lib.rs]: Did you intent to change this from `"tidy error: "` to `"some tidy checks failed"`?

Comment (frewsxcv) [reply src/tools/tidy/src/lib.rs]: Nope, good catch! Fixed in the latest force push

**Comment by shepmaster** [comment]
Hey, @nrc, will you have a chance to get to this soon?

**Comment by shepmaster** [comment]
A birdie told me that @nrc is on leave for a bit, so let's pick someone from the libs team... r? @Kimundi

**Comment by michaelwoerister** [comment]
@frewsxcv Did you test if the instances that end up in `rustc` actually print something? I never had any luck with that. I suspect that `libterm` somehow interferes with stderr. Then again, looking at the changes, it probably didn't work before either in those cases.

**Pull request merged**